src/adstaylorinteg.jl

```diff
@@ -0,0 +1,326 @@
+                        view(ys, i, idxs), p0; inplace = true)
+        # Estimate next order coefficient
+        exp_model!(view(M, i:i), varorder+1, fit.param)
+    end
+
+    return M
+end
+
+function size_per_variable(P::AbstractVector{TaylorN{T}}) where {T <: Real}
+    # Number of variables
+    nv = get_numvars()
+    # Size per variable per element of P
+    norms = Matrix{T}(undef, nv, length(P))
+    for i in eachindex(P)
+        norms[:, i] .= size_per_variable(P[i])
+    end
+    # Sum over element of P
+    return sum(norms, dims = 2)
+end
+
+"""
+    splitdirection(P::TaylorN{T}) where {T <: Real}
+    splitdirection(P::AbstractVector{TaylorN{T}}) where {T <: Real}
+
+Return the index of the jet transport variable with the largest truncation error
+according to [`size_per_variable`](@ref).
+"""
+function splitdirection(P::TaylorN{T}) where {T <: Real}
+    # Size per variable
+    M = size_per_variable(P)
+    # Variable with maximum error
+    return argmax(M)[1]
+end
+
+function splitdirection(P::AbstractVector{TaylorN{T}}) where {T <: Real}
+    # Size per variable
+    M = size_per_variable(P)
+    # Variable with maximum error
+    return argmax(M)[1]
+end
+
+"""
+    adsnorm(P::TaylorN{T}) where {T <: Real}
+
+Return the truncation error of `P`.
+"""
+function adsnorm(P::TaylorN{T}) where {T <: Real}
+    # Jet transport order
+    varorder = P.order
+    # Absolute sum per order
+    ys = Vector{T}(undef, varorder+1)
+
+    for i in eachindex(ys)
+        ys[i] = sum(abs, P.coeffs[i].coeffs)
+    end
+    # Initial parameters
+    p0 = ones(T, 2)
+    # Orders
+    xs = 0:varorder
+    # Non zero coefficients
+    idxs = findall(!iszero, ys)
+    # Fit exponential model
+    fit = curve_fit(exp_model!, exp_model_jacobian!, view(xs, idxs),
+                        view(ys, idxs), p0; inplace = true)
+    # Estimate next order coefficient
+    return exp_model(varorder+1, fit.param)
+end
+
+# Split node's domain in half
+# See section 3 of https://doi.org/10.1007/s10569-015-9618-3
+function split(node::ADSBinaryNode{N, M, T}, x::SVector{M, TaylorN{T}},
+               p::SVector{M, Taylor1{TaylorN{T}}}) where {N, M, T <: Real}
+    # Split direction
+    j = splitdirection(x)
+    # Split domain
+    s1, s2 = split(node.s, j)
+    # Jet transport variables
+    v_1, v_2 = get_variables(), get_variables()
+    # Shift expansion point
+    r = getroot(node).s
+    v_1[j] = v_1[j]/2 + r.lo[j]/2
+    v_2[j] = v_2[j]/2 + r.hi[j]/2
+    # Taylor1 order
+    order = p[1].order
+    # Left half
+    x1 = SVector{M, TaylorN{T}}(x[i](v_1) for i in eachindex(x))
+    p1 = SVector{M, Taylor1{TaylorN{T}}}(
+        Taylor1( map(z -> z(v_1), p[i].coeffs), order ) for i in eachindex(p)
+    )
+    # Right half
+    x2 = SVector{M, TaylorN{T}}(x[i](v_2) for i in eachindex(x))
+    p2 = SVector{M, Taylor1{TaylorN{T}}}(
+        Taylor1( map(z -> z(v_2), p[i].coeffs), order ) for i in eachindex(p)
+    )
+
+    return s1, x1, p1, s2, x2, p2
```

Comment:
If we passed `dt` as an argument to `split`, then here we could return a tuple of `ADSBinaryNode`s, i.e.,
```julia
    return ADSBinaryNode(s1, node.t+dt, x1, p1), ADSBinaryNode(s2, node.t+dt, x2, p2)
```
and then set those below in `split!` in the corresponding calls to `leftchild!` and `rightchild!`; though I think it'd require defining new methods:

```julia
function leftchild!(parent::ADSBinaryNode, newnode::ADSBinaryNode)
    # set `newnode` as left child of `parent`
end
```

Exploiting the type parametrization in the `ADSBinaryNode` constructor can help with type-stability.

Reply:
Naive question: is there an issue with type stability?